### PR TITLE
Fix simulator borking on setting contact phone number

### PR DESCRIPTION
**Pull request by nicpottier**

### CODE REVIEW

**Comment by ericnewcomer** [comment]
:hamburger:

**Comment by nicpottier** [comment]
review again plz.

File: temba/flows/tests.py [hunk 1]
```diff
@@ -1244,6 +1244,9 @@ def test_save_to_contact_action(self):
         sms = self.create_msg(direction=INCOMING, contact=self.contact,
                               text="+12065551212", contact_urn=contact.urns.filter(path='+250788382382').first())
 
+        # create another contact with that phone number, to test stealing
+        robbed = self.create_contact("Robzor", "+12065551212")
```
Comment (ericnewcomer) [inline temba/flows/tests.py]: cute.

**Comment by ericnewcomer** [comment]
Curious what happens when we edit a contact that has two phone numbers from the contact page. How does it decide which number to show there? Top priority phone? Does it remove the same one when editing it? I presume we show the top priority phone on contact list pages as well.

**Comment by nicpottier** [comment]
It shows the top priority phone.

**Comment by ericnewcomer** [comment]
Right, but editing does it remove the top priority phone as well?

File: temba/contacts/models.py [hunk 2]
```diff
@@ -1050,7 +1056,9 @@ def update_urns(self, urns):
                     urn.save()
                     urns_attached.append(urn)
                 elif urn.contact != self:
```
Comment (norkans7) [inline temba/contacts/models.py]: probably this block can be unified with the above one

File: temba/contacts/models.py [hunk 3]
```diff
@@ -994,6 +994,12 @@ def set_first_name(self, first_name):
             names = [first_name] + names[1:]
             self.name = " ".join(names)
 
+    def get_urns_for_scheme(self, scheme):
+        """
+        Returns all the URNs for the passed in scheme
+        """
+        return self.urns.filter(scheme=scheme).order_by('-priority', 'pk')
```
Comment (norkans7) [inline temba/contacts/models.py]: Here I think ordering by modified_on so that the older urn always stay at the begining of the list the same in get_urns

Comment (nicpottier) [inline temba/contacts/models.py]: We actually want the ordering to be priority all the time, not most frequent, as that will be the URN messages are sent to.

Comment (norkans7) [inline temba/contacts/models.py]: yep I think something like ('-priority', 'modified_on', 'pk') really I think contact.get_urn should stay consistent as long as the urn still belongs to the contact

Comment (nicpottier) [inline temba/contacts/models.py]: URNs don't have a modified on.. so pk is the best we can do here.

Comment (norkans7) [inline temba/contacts/models.py]: Right,I did not checked that

**Pull request merged**